src/box/memtx_tree.cc

```diff
@@ -420,33 +437,30 @@ tree_iterator_prev_equal_base(struct iterator *iterator, struct tuple **ret)
 	}
 	memtx_tree_iterator_prev(&index->tree, &it->tree_iterator);
 	struct tuple *successor = it->current.tuple;
+	tuple_ref(successor);
 	struct memtx_tree_data<USE_HINT> *res =
 		memtx_tree_iterator_get_elem(&index->tree, &it->tree_iterator);
+	struct index *idx = iterator->index;
+	struct space *space = space_by_id(iterator->space_id);
 	/* Use user key def to save a few loops. */
 	if (res == NULL ||
 	    tuple_compare_with_key(res->tuple, res->hint,
 				   it->key_data.key,
 				   it->key_data.part_count,
 				   it->key_data.hint,
 				   index->base.def->key_def) != 0) {
+		tree_iterator_set_current<USE_HINT>(it, NULL);
```

Comment:
May be use tree_iterator_set_current_tuple here to prevent changing hint as it was previously

Reply:
See my [reply](https://github.com/tarantool/tarantool/pull/6798/commits/714186ed47ee00fd7d1af5b32c66bbfa2e745004#r794646568) in the different thread.